qubesmanager/qube_manager.py

```diff
@@ -239,10 +243,16 @@ def update_power_state(self):
                     except exc.QubesDaemonAccessError:
                         pass
 
-            if self.vm.klass in {'TemplateVM', 'StandaloneVM'} and \
-                    manager_utils.get_feature(
+            if self.vm.klass in {'TemplateVM', 'StandaloneVM'}:
+                if manager_utils.get_feature(
                         self.vm, 'updates-available', False):
-                self.state['outdated'] = 'update'
+                    self.state['outdated'] = 'update'
+                elif manager_utils.get_feature(
+                        self.vm, 'os-eol', None):
+                    d = [int(x) for x in manager_utils.get_feature(self.vm, 'os-eol', '0-0-0').split('-')]
+                    eol = datetime(d[0], d[1], d[2])
```

Comment:
```suggestion
                        eol_string: str = vm.features.get('os-eol', '')
                        eol = datetime.strptime(eol_string, '%Y-%m-%d')
```

This is a bit more readable, but more importantly, it's the same logic as in the qui updater widget, to be sure we use the same one - that's the function https://github.com/QubesOS/qubes-desktop-linux-manager/blob/8bf07aa132dd47d1b501ec4ec2706c042f77dc7e/qui/utils.py#L88

Reply:
Applied it and tested it. With one change that it should be `self.vm.features.get`

Reply:
ah, right, you're correct